zipkin-server/src/main/resources/zipkin-server-shared.yml

```diff
@@ -143,14 +143,24 @@ zipkin:
     base-path: /zipkin
 
 server:
-  port: ${QUERY_PORT:9411}
   use-forward-headers: true
   compression:
     enabled: true
     # compresses any response over min-response-size (default is 2KiB)
     # Includes dynamic json content and large static assets from zipkin-ui
     mime-types: application/json,application/javascript,text/css,image/svg
 
+# Disable the Web Server.
+# See https://docs.spring.io/spring-boot/docs/current/reference/html/howto-embedded-web-servers.html.
+spring.main.web-application-type: none
+
+# Additional server ports specified with Armeria configuration.
+armeria:
+  ports:
+    - port: ${QUERY_PORT:${server.port:9411}}
```

Comment:
oooo nesting...

Reply:
I'm not sure but isn't it enough to specify `${QUERY_PORT:9411}`?